File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        
        string target = "";
        for(int i=0; i<s.size(); i++){
            if(isalpha(s[i]) || isdigit(s[i])){
                target += tolower(s[i]);
            }
        }
        // cout << target << endl;
        
        if(target.size() == 0){
            return true;
        }
        
        for(int i=0; i<target.size()/2 + 1; i++){
            if(target[i] != target[target.size() - 1 - i]){
                return false;
            }
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String target = "";
        for(int i=0; i<s.length(); i++){
            if(Character.isLetter(s.charAt(i)) || Character.isDigit(s.charAt(i))){
                target += Character.toLowerCase(s.charAt(i));
            }
        }
        
        // System.out.println(target);
        
        if(target.length() == 0){
            return true;
        }

        for(int i=0; i<target.length()/2 + 1; i++){
            if(target.charAt(i) != target.charAt(target.length() - 1 - i)){
                return false;
            }
        }
        return true;
        
    }
}